### PR TITLE
Use GKE managed prow certs

### LINKED ISSUE #14944: The cert-manager version we are using is not supported by letsencrypt.
> We've been working with Jetstack, the authors of cert-manager, on a
> series of fixes to the client. Cert-manager sometimes falls into a
> traffic pattern where it sends really excessive traffic to Let's
> Encrypt's servers, continuously. To mitigate this, we plan to start
> blocking all traffic from cert-manager versions less than 0.8.0 (the
> current semver minor release), as of November 1, 2019. Please upgrade
> all of your cert-manager instances before then.
> 
> We're sending this email because this is the contact address of your
> cert-manager instance at:
> 
>  35.185.195.73 .
> 
> Version 0.8.0 is much better but we still observe excessive traffic in
> some cases. We're working with Jetstack to improve these cases. As new
> versions of cert-manager are released, we will add the non-current
> versions to our block list after 3 months. We strongly encourage
> cert-manager users to stay up-to-date with new versions.
> 
> You can subscribe to new version notifications by going to cert-manager's
> GitHub project (https://github.com/jetstack/cert-manager), clicking
> "Watch" in the upper right, then clicking "Releases only."
> 
> Also, there is an opportunity to help both Jetstack and Let's Encrypt.
> Once you've upgraded, please check the logs for your cert-manager
> instances from time to time. Are they making excessive requests to Let's
> Encrypt (more than, say, 10 per day over multiple days)? If so, please
> share details at https://github.com/jetstack/cert-manager/issues/1948 .
> 
> Thanks,
> Let's Encrypt Team

We are currently using version `v0.5.2`: https://github.com/kubernetes/test-infra/blob/7abdd3fbe9022f7adda942b987d4c9ce47bd5460/prow/cluster/cert-manager.yaml#L155

Unfortunately there are backwards incompatible changes that make upgrading cert-manager more complicated than just changing the image version: https://docs.cert-manager.io/en/latest/tasks/upgrading/

The current certificate for prow.k8s.io is good until Monday, January 6, 2020 at 11:47:17 AM

/kind oncall-hotlist
@Katharine @fejta @BenTheElder @stevekuznetsov 

I know we have discussed using Google managed SSL certificates for prow.k8s.io. If we want to make that change now would be a good time.

### PR DESCRIPTION
**Pull request by Katharine**
This fixes #14944 and removes some operational complexity from us.

Actually deploying it as is would cause some downtime (10-20 minutes?) while the certificate was created by GKE, and it's hard to test that migration. It may be possible to convince GKE to generate the certificate while still serving the cert-manager one; I'm not sure.

I prefer this over going through upgrading cert-manager, though.

/cc @cjwagner @fejta @stevekuznetsov @BenTheElder

### CODE REVIEW

File: prow/cluster/BUILD.bazel [hunk 1]
```diff
@@ -32,8 +32,6 @@ release(
     ),
     component("build", "deployment"),
     component("build_rbac", MULTI_KIND),
-    component("cert-manager", MULTI_KIND),
```
Comment (fejta) [inline prow/cluster/BUILD.bazel]: Remember that'll we'll need to manually `kubectl delete -f cert-manager.yaml`

**Comment by cjwagner** [comment]
Could we set up the managed certificate on a new ingress that initially only serves a dummy URI path we don't care about so that the existing ingress still handles all real traffic? Then once we've confirmed that the new certificate has been provisioned by checking the dummy path we can update the ingress to serve all paths and delete the old ingress.

**Comment by Katharine** [comment]
@cjwagner I don't think so - the verification requires the GCLB to be accessible at the expected domain, or the certificate won't be generated.

I can imagine we can partially configure it such that the verification is performed while still serving using the old secret, but I'm not sure how the ingress will handle the conflict once it produces its own cert

**Comment by fejta** [comment]
Couldn't we have a prow-canary.k8s.io that uses the new path? And then once this works we can tell people worried about the brief outage to use this URL while we make prow.k8s.io use the new path as well?

**Comment by cjwagner** [comment]
> I can imagine we can partially configure it such that the verification is performed while still serving using the old secret

Yeah thats what I mean to suggest.

> but I'm not sure how the ingress will handle the conflict once it produces its own cert.

Would there be a conflict if the ingresses don't handle overlapping paths? Having multiple ingresses handle different paths of the same domain is supported and TLS termination is configured per ingress so it seems like this could be possible.

Definitely agree that it is unclear if this would work though and we'd probably want to demo it first to find out.

**Comment by Katharine** [comment]
> Couldn't we have a prow-canary.k8s.io that uses the new path? And then once this works we can tell people worried about the brief outage to use this URL while we make prow.k8s.io use the new path as well?

Sorry, yes, that would work - I misread Cole's comment.

**Comment by cjwagner** [comment]
Using a separate domain sounds more straightforward and less likely to encounter problems.

**Comment by Katharine** [comment]
This has now been completed in #15726, #15727 and #15728.

**Pull request closed without merging**